mdoc/Mono.Documentation/Updater/Formatters/FSharpFormatter.cs

```diff
@@ -0,0 +1,1084 @@
+        protected HashSet<int> GetCurryBorders(MethodDefinition method)
+        {
+            var compilationArgumentCounts = GetCustomAttribute(method.CustomAttributes,
+                "CompilationArgumentCountsAttribute");
+
+            int[] curryCounts = { };
+            if (compilationArgumentCounts != null)
+            {
+                var customAttributeArguments =
+                    (CustomAttributeArgument[])compilationArgumentCounts.ConstructorArguments[0].Value;
+                curryCounts = customAttributeArguments.Select(i => (int)i.Value).ToArray();
+            }
+            var curryBorders = new HashSet<int>();
+            int sum = 0;
+            foreach (var curryCount in curryCounts)
+            {
+                sum += curryCount;
+                curryBorders.Add(sum);
+            }
+            return curryBorders;
+        }
+
+        private CustomAttribute GetCustomAttribute(Collection<CustomAttribute> customAttributes, string name)
+        {
+            return customAttributes.SingleOrDefault(i => i.AttributeType.Name == name);
+        }
+
+        protected bool TryAppendOperatorName(StringBuilder buf, MethodDefinition method)
+        {
+            if (!IsOperator(method))
+                return false;
+            if (operators.ContainsKey(method.Name))
+            {
+                buf.Append($"( {operators[method.Name]} )");
+                return true;
+            }
+
+            if (TryAppendCombinatedOperandName(buf, method))
+                return true;
+
+            return false;
+        }
+
+        private static bool TryAppendCombinatedOperandName(StringBuilder buf, MethodDefinition method)
+        {
+            var oldName = method.Name.Remove(0, 3);
+            var newName = new StringBuilder();
+            bool found;
+            do
+            {
+                found = false;
+                foreach (var op in combinatedOperators)
+                {
+                    if (oldName.StartsWith(op.Key))
+                    {
+                        oldName = oldName.Remove(0, op.Key.Length);
+                        newName.Append(op.Value);
+                        found = true;
+                    }
+                }
+            } while (found);
+
+            if (oldName.Length == 0)
+            {
+                buf.Append($"( {newName} )");
+                return true;
+            }
+            return false;
+        }
+        
+        protected override StringBuilder AppendPointerTypeName(StringBuilder buf, TypeReference type, DynamicParserContext context)
+        {
+            TypeSpecification spec = type as TypeSpecification;
+            buf.Append("nativeptr<");
+            _AppendTypeName(buf, spec != null ? spec.ElementType : type.GetElementType(), context);
+            buf.Append(">");
+            return buf;
+        }
+
+        private static string GetTab()
```

Comment:
It would _probably_ be a bit cleaner to remove this method, and simply add a const field to the `Consts` class ... say, `Consts.Tab`. That way, it can be used any other place where it's needed, rather than calling a method that returns a new instance of  a string.